make/build.js

```diff
@@ -121,6 +121,26 @@ module.exports = function build(settings, task) {
     });
 
     //////////////////////////////////////////////////////////////////////
+    // Make sure all examples compile
+    //////////////////////////////////////////////////////////////////////
+
+    task("excomp", [], function() {
+        this.excomp(
+            "examples/**/*.html",
+            "src/js/libcs/Parser.js",
+            function(html, parser) {
+                var re = /<script[^>]*type *= *['"]text\/x-cindyscript['"][^>]*>([^]*?)<\/script>/g;
```

Comment:
No, the `[^]` is intentional; it's meant to match any character at all, including newlines. JavaScript uses `/foo[^]bar/` where other languages might be using `/foo.bar/s` to make the `.` match newlines as well.

I'm using the non-greedy quantifier `*?`, which matches the shortest possible string. So as long as the script tags are properly formatted, this will ensure that we don't capture portions of the next script. Looking at [the HTML spec for end tags](https://html.spec.whatwg.org/multipage/syntax.html#end-tags), the tag name might be using a different case, and there might be whitespace between that and the closing `>`. We could adjust the regexp for that, but I think that the way it is, we'd very likely get a false positive instead, since the combined code from both scripts and including the tags between them is unlikely to be valid CindyScript. So that error would just remind us to format the `</script>` properly, which [we need for the website](https://github.com/CindyJS/website/blob/4732edc4eb54d24d6942ec22088d528c4886d0d5/lib/examples.js#L29), too.